metadata-ingestion/src/datahub/ingestion/source/bigquery.py

```diff
@@ -1,14 +1,32 @@
-from typing import Optional
+from typing import Optional, Tuple
 
 from .sql_common import SQLAlchemyConfig, SQLAlchemySource
 
 
 class BigQueryConfig(SQLAlchemyConfig):
     scheme = "bigquery"
-    project_id: Optional[str]
+    project_id: Optional[str] = None
 
     def get_sql_alchemy_url(self):
-        return f"{self.scheme}://{self.project_id}"
+        if self.project_id:
+            return f"{self.scheme}://{self.project_id}"
+        # When project_id is not set, we will attempt to detect the project ID
+        # based on the credentials or environment variables.
+        # See https://github.com/mxmzdlv/pybigquery#authentication.
+        return f"{self.scheme}://"
+
+    def get_identifier(self, schema: str, table: str) -> str:
+        if self.project_id:
+            return f"{self.project_id}.{schema}.{table}"
+        return f"{schema}.{table}"
+
+    def mangle_schema_table_names(self, schema: str, table: str) -> Tuple[str, str]:
```

Comment:
Needs a comment here... describing why we need to do this for BigQuery

Reply:
added